modules/random/include/hephaestus/random/random_object_creator.h

```diff
@@ -200,4 +211,65 @@ template <RandomCreatableVector T>
   return vec;
 };
 
+//=================================================================================================
+// Random array creation
+//=================================================================================================
+template <typename T>
+concept RandomCreatableArray = ArrayType<T> && RandomCreatable<typename T::value_type>;
+
+/// Fill a vector with randomly generated values of type T.
+template <RandomCreatableArray T>
+[[nodiscard]] auto random(std::mt19937_64& mt) -> T {
+  T array;
+  const auto size = array.size();
+
+  for (std::size_t it = 0; it < size; ++it) {
+    array[it] = random<typename T::value_type>(mt);
+  }
```

Comment:
I think you can actually write this as:
```suggestion
  T array;
  std::ranges::generate(array, [&mt](){random<typename T::value_type>(mt)});
```